### ISSUE TITLE
asSchema should have Product as upper type bound 

### ISSUE THREAD

**Issue by manuzhang** [MrPowers/spark-daria#78]
Careless user like me may leave `()` out when defining the schema

```scala
val df = spark.createDF(List("foo", "bar"),  List("data", StringType, true))
```

Currently, a `MatchError` is thrown at runtime.
Would it be better to report error at compile time ? 

@MrPowers

**Comment by MrPowers** [comment]
@manuzhang - Yea, I think we should try to throw compile time errors instead of runtime errors whenever possible.  Do you have any ideas how to improve this with a PR?  If so, please send it over!

**Comment by MrPowers** [comment]
@snithish - FYI, I know this code if your baby ;)

Pretty awesome that people are the code you wrote every day.  I use the `createDF` function all the time.

**Comment by manuzhang** [comment]
@MrPowers  Opened https://github.com/MrPowers/spark-daria/pull/81

**Comment by manuzhang** [comment]
Fixed in #81
